src/util/coordinateconversion.cpp

```diff
@@ -24,26 +24,298 @@
+            );
+            return false;
+        }
+
+        return true;
+    }
+} // namespace
+
 namespace openspace {
 
-glm::dvec3 icrsToGalacticCartesian(float ra, float dec, double distance) {
-    // Convert to Galactic Coordinates from ICRS right ascension and declination
-    // https://gea.esac.esa.int/archive/documentation/GDR2/Data_processing/
-    // chap_cu3ast/sec_cu3ast_intro/ssec_cu3ast_intro_tansforms.html#SSS1
-    const glm::dmat3 conversionMatrix = glm::dmat3({
-        -0.0548755604162154,  0.4941094278755837, -0.8676661490190047, // col 0
-        -0.8734370902348850, -0.4448296299600112, -0.1980763734312015, // col 1
-        -0.4838350155487132,  0.7469822444972189,  0.4559837761750669  // col 2
-    });
-
-    glm::dvec3 rICRS = glm::dvec3(
-        cos(glm::radians(ra)) * cos(glm::radians(dec)),
-        sin(glm::radians(ra)) * cos(glm::radians(dec)),
-        sin(glm::radians(dec))
+// Convert Equatorial coordinates ICRS right ascension and declination (a, d)
+// into Galactic coordinates (l, b)
+glm::dvec3 icrsToGalacticCartesian(double ra, double dec, double distance,
+                                   bool isDegrees)
+{
+    // Reference:
+    // https://www.atnf.csiro.au/people/Tobias.Westmeier/tools_coords.php
+
+    // (Ra, Dec) -> (a, d)
+    double a = isDegrees ? glm::radians(ra) : ra;
+    double d = isDegrees ? glm::radians(dec) : dec;
+
+    // Convert to galactic reference frame
+    double l = L0 - atan2(
+        cos(d) * sin(a - A0),
+        sin(d) * cos(D0) - cos(d) * sin(D0) * cos(a - A0)
+    );
+    double b = asin(sin(d) * sin(D0) + cos(d) * cos(D0) * cos(a - A0));
+
+    // Convert to cartesian
+    glm::dvec3 rGalactic = glm::dvec3(
+        cos(b) * cos(l),
+        cos(b) * sin(l),
+        sin(b)
     );
-    glm::dvec3 rGalactic = conversionMatrix * rICRS; // on the unit sphere
 
     return distance * rGalactic;
 }
 
+// Ra format 'XhYmZs', where X and Y are positive integers and Z is a positive double
+// Dec format 'XdYmZs', where X is a signed integer, Y is a positive integer and Z is a
+// positive double
+glm::dvec2 icrsToDecimalDegrees(const std::string& ra, const std::string& dec) {
+    // Reference:
+    // https://math.stackexchange.com/questions/15323/how-do-i-calculate-the-cartesian-coordinates-of-stars
+    if (ra.size() < 6 || dec.size() < 6) {
+        throw(ghoul::lua::LuaRuntimeException(fmt::format(
+            "Ra '{}' or Dec '{}' format is incorrect. Correct format is: Ra 'XhYmZs', "
+            "and Dec 'XdYmZs'", ra, dec)
+        ));
+    }
+
+    // Parse right ascension
+    int ra_hours, ra_minutes;
+    double ra_seconds;
+    parseRa(ra, ra_hours, ra_minutes, ra_seconds);
+
+    // Parse declination
+    int dec_degrees, dec_minutes;
+    double dec_seconds;
+    parseDec(dec, dec_degrees, dec_minutes, dec_seconds);
+
+    if (!isRaDecValid(ra_hours, ra_minutes, ra_seconds, dec_degrees, dec_minutes,
```

Comment:
This might be better to write:
```
const bool isValid = isRaDecValid(
  raHours,
  raMinutes,
  raSeconds,
  decDegrees,
  decMinutes,
  decSeconds
);

if (!isValid) {
```

Otherwise the `if` statement would be lost when splitting the function call into multiple lines